src/firmware/app/control/trajectory_planner.c

```diff
@@ -1,215 +1,222 @@
-                                                        arc_length_parameterization);
-
-    const float arc_segment_length =
-        shared_polynomial2d_getTotalArcLength(arc_length_parameterization) / num_segments;
-
-    // Now use numerical interpolation to get constant arc length segments for the
-    // parameterization
-    app_trajectory_planner_generateConstArclengthTrajectoryPositions(
-        traj_elements, path, num_segments, arc_length_parameterization,
-        arc_segment_length);
-
-    // Generate the profile for the maximum physically valid velocity at each point on the
-    // profile
-    float max_allowable_speed_profile[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
-    app_trajectory_planner_getMaxAllowableSpeedProfile(
-        max_allowable_speed_profile, path, num_segments, arc_length_parameterization,
-        arc_segment_length, max_allowable_acceleration);
-
-    float* velocity_profile = trajectory->speed_profile;
-    // First point on the velocity profile is always the current speed
-    velocity_profile[0] = initial_speed;
-
-    // Loop through the path forwards to ensure that the maximum velocity limit defined by
-    // the curvature is not breached by constant max acceleration of the robot (if it is,
-    // pull down the acceleration)
-    app_trajectory_planner_generateForwardsContinuousVelocityProfile(
-        num_segments, velocity_profile, max_allowable_speed_profile, arc_segment_length,
-        max_allowable_acceleration, max_allowable_speed);
-
-    // Check that it was physically possible for the robot to reach the final velocity
-    // requested
-    if (velocity_profile[num_segments - 1] >= final_speed)
+    // Create a 2d polynomial out of the theta profile and a polynomial of all zeros
+    Polynomial2dOrder3_t theta_poly_2d = {.x = theta_poly, .y = {0, 0, 0, 0}};
+    app_trajectory_planner_getMaximumSpeedProfile_impl(
+        theta_poly_2d, num_elements, t_start, t_end, max_angular_acceleration,
+        max_angular_speed, max_allowable_speed_profile);
+
+    // The initial and final angular velocity of a path is assumed to be zero
+    const float initial_angular_speed = 0;
+    const float final_angular_speed   = 0;
+    // Generate the forwards continuous angular speed profile
+    status = app_trajectory_planner_createForwardsContinuousSpeedProfile_impl(
+        num_elements, angular_segment_lengths, max_allowable_speed_profile,
+        max_angular_acceleration, initial_angular_speed, final_angular_speed,
+        angular_speed);
+    if (status != OK)
     {
-        velocity_profile[num_segments - 1] = final_speed;
+        return status;
     }
-    else
+
+    // Make the speed profiles backwards continuous
+    status = app_trajectory_planner_modifySpeedsToBeBackwardsContinuous_impl(
+        num_elements, linear_segment_lengths, max_linear_acceleration,
+        initial_linear_speed, linear_speed);
+    if (status != OK)
```

Comment:
Gareth requested this for optimization purposes, I do see the potential problem here. I'll remove the early return now. We'll need to do a new PR for optimization anyways because there are no values returned RN. @garethellis0